app/src/main/java/org/jtaccuino/app/persistence/FilePersistence.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2025 JTaccuino Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jtaccuino.app.persistence;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FilePersistence {
+
+    private static FilePersistence INSTANCE = PersistenceManager.readPersistenceFile("files", FilePersistence.class).orElse(new FilePersistence());
+
+    public static record OpenFile(Path path) {
+    }
+
+    public static record RecentFile(Path path) {
+    }
+
+    public static FilePersistence getDefault() {
+        return INSTANCE;
+    }
+
+    public List<OpenFile> openFiles = new ArrayList<>();
```

Comment:
public non-final field but with modifier methods in the same class.
please implement a more defensive storage. defensive copies and private access and all the best practices...